src/micromamba/common_options.cpp

```diff
@@ -0,0 +1,489 @@
+                   retry_clean_cache.set_cli_config(false),
+                   retry_clean_cache.description())
+        ->group(cli_group);
+}
+
+
+void
+init_channel_parser(CLI::App* subcom)
+{
+    auto& config = Configuration::instance();
+
+    auto& channels = config.at("channels").get_wrapped<std::vector<std::string>>();
+    channels.set_post_build_hook(channels_hook);
+    subcom->add_option("-c,--channel", channels.set_cli_config({}), channels.description())
+        ->type_size(1)
+        ->allow_extra_args(false);
+
+    auto& override_channels = config.insert(Configurable("override_channels", false)
+                                                .group("cli")
+                                                .rc_configurable(false)
+                                                .set_env_var_name()
+                                                .description("Override channels")
+                                                .set_post_build_hook(override_channels_hook));
+    subcom->add_flag("--override-channels",
+                     override_channels.set_cli_config(false),
+                     override_channels.description());
+
+    auto& channel_priority = config.at("channel_priority").get_wrapped<ChannelPriority>();
+    subcom->add_set("--channel-priority",
+                    channel_priority.set_cli_config(""),
+                    { "strict", "disabled" },
+                    channel_priority.description());
+
+    auto& strict_channel_priority
+        = config.insert(Configurable("strict_channel_priority", false)
+                            .group("cli")
+                            .rc_configurable(false)
+                            .description("Enable strict channel priority")
+                            .set_post_build_hook(strict_channel_priority_hook));
+    subcom->add_flag("--strict-channel-priority",
+                     strict_channel_priority.set_cli_config(0),
+                     strict_channel_priority.description());
+
+    auto& no_channel_priority = config.insert(Configurable("no_channel_priority", false)
+                                                  .group("cli")
+                                                  .rc_configurable(false)
+                                                  .description("Disable channel priority")
+                                                  .set_post_build_hook(no_channel_priority_hook));
+    subcom->add_flag("--no-channel-priority",
+                     no_channel_priority.set_cli_config(0),
+                     no_channel_priority.description());
+}
+
+void
+channels_hook(std::vector<std::string>& channels)
+{
+    auto& config = Configuration::instance();
+
+    // Workaround to silent the hook warning
+    // TODO: resolve configurable deps/workflow to compute only once
+    auto verbosity = MessageLogger::global_log_severity();
+    MessageLogger::global_log_severity() = mamba::LogSeverity::error;
```

Comment:
these enums have not been adjusted to the new kError format?

Reply:
done